src/openloco/windows/KeyboardShortcuts.cpp

```diff
@@ -0,0 +1,202 @@
+#include "../ui/WindowManager.h"
+
+using namespace openloco::interop;
+using namespace openloco::input;
+
+namespace openloco::ui::KeyboardShortcuts
+{
+
+    static const int rowHeight = 10;
+
+    static window_event_list _events;
+    static loco_global<char[16], 0x0112C826> _commonFormatArgs;
+
+    static widget_t _widgets[] = {
+        make_widget({ 0, 0 }, { 360, 238 }, widget_type::frame, 0),
+        make_widget({ 1, 1 }, { 358, 13 }, widget_type::caption_25, 0, string_ids::keyboard_shortcuts),
+        make_widget({ 345, 2 }, { 13, 13 }, widget_type::wt_9, 0, image_ids::close_button, string_ids::tooltip_close_window),
+        make_widget({ 0, 15 }, { 360, 223 }, widget_type::panel, 1),
+        make_widget({ 4, 19 }, { 352, 202 }, widget_type::scrollview, 1, vertical, string_ids::keyboard_shortcut_list_tip),
+        make_widget({ 4, 223 }, { 150, 12 }, widget_type::wt_11, 1, string_ids::reset_keys, string_ids::reset_keys_tip),
+        widget_end(),
+    };
+
+    namespace widx
+    {
+        enum
+        {
+            frame,
+            caption,
+            close_button,
+            panel,
+            list,
+            customize_keys_btn,
+        };
+    }
+
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi);
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex);
+    static void on_mouse_up(window* self, widget_index i);
+    static void loc_4BE832(window* self);
+    static void tooltip(window*, widget_index);
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight);
+    static void on_scroll_mouse_over(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index);
+    static void on_scroll_mouse_down(ui::window* self, int16_t x, int16_t y, uint8_t scroll_index);
+
+    static void init_events()
+    {
+        _events.on_mouse_up = on_mouse_up;
+        _events.get_scroll_size = get_scroll_size;
+        _events.scroll_mouse_down = on_scroll_mouse_down;
+        _events.scroll_mouse_over = on_scroll_mouse_over;
+        _events.tooltip = tooltip;
+        _events.draw = draw;
+        _events.draw_scroll = draw_scroll;
+    }
+
+    // 0x004BE6C7
+    window* open()
+    {
+        window* window;
+
+        window = WindowManager::bringToFront(WindowType::keyboardShortcuts, 0);
+        if (window != nullptr)
+            return window;
+
+        init_events();
+
+        // 0x004BF833 (create_options_window)
+        window = WindowManager::createWindowCentred(WindowType::keyboardShortcuts, { 360, 238 }, 0, &_events);
+
+        window->widgets = _widgets;
+        window->enabled_widgets = (1 << widx::close_button) | (1 << widx::customize_keys_btn);
+        window->init_scroll_widgets();
+
+        auto skin = objectmgr::get<interface_skin_object>();
+        window->colours[0] = skin->colour_0B;
+        window->colours[1] = skin->colour_10;
+
+        window->row_count = 35;
```

Comment:
Could you introduce a getter in ShortcutManager to get this number, so we have a central place to manage the shortcuts?